src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1507,7 +1344,8 @@ BasicType java_lang_Class::primitive_type(oop java_class) {
   } else {
     assert(java_class == Universe::void_mirror(), "only valid non-array primitive");
   }
-  assert(Universe::java_mirror(type) == java_class, "must be consistent");
+  assert(Universe::java_mirror(type) == java_class ||
```

Comment:
If we need the condition `HeapShared::scratch_java_mirror(type) == java_class` here, shouldn't the previous assert be also updated to check for scratch java mirror for void type?

  `assert(java_class == Universe::void_mirror() |, "only valid non-array primitive");`